core/drivers/crypto/caam/hal/common/hal_rng.c

```diff
@@ -62,6 +62,12 @@ bool caam_hal_rng_key_loaded(vaddr_t baseaddr)
 	return io_caam_read32(baseaddr + RNG_STA) & RNG_STA_SKVN;
 }
 
+bool __weak caam_hal_rng_pr_enabled(vaddr_t baseaddr)
+{
+	return (io_caam_read32(baseaddr + RNG_STA) &
+		(RNG_STA_PR0 | RNG_STA_PR1)) == (RNG_STA_PR0 | RNG_STA_PR1);
```

Comment:
You may not always have 2 state handles, depending on the platform.

Reply:
As far as OP-TEE is concerned, don't we always have two handles, because `RNG_DATABUF_NB == 2`?